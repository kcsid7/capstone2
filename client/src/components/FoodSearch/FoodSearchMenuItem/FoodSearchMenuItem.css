.MenuItem {
    margin: 1rem 3rem;
    width: 100%;
    min-height: 175px;
    padding: 1rem 2rem;
    box-shadow: -2px -2px 2px darkslategray, 2px 2px 2px darkslategray;
}
 
.MenuItem-Image {
    margin-right: auto;
    display: flex;
    justify-content: center;
}

.MenuItem-Image-Img {
    height: 150px;
    width: 150px;
    align-self: center;
    box-shadow: 7px 7px 7px darkslategrey;
}
 
.MenuItem-Buttons {
    display: flex;
    justify-content: center;
    align-items: center;
}

.MenuItem-Details {
    width: 50%;
    margin: 1rem 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
}