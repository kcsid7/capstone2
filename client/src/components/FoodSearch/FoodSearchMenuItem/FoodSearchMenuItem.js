import React, { useState, useContext, useEffect } from "react";

import { Link } from "react-router-dom";

// CSS
import "./FoodSearchMenuItem.css"

function FoodSearchMenuItem(props) {

   let menuItemImage = props.image;

    return (
        <>
        <div className="MenuItem container row">
            <div className={`MenuItem-Image col-12 col-sm-12 col-xl-2`}>
                {
                    menuItemImage ?
                    <img 
                        className="MenuItem-Image-Img" 
                        src={menuItemImage}/>
                    : ""
                }
            </div>
            <div className={`MenuItem-Details col-12 col-sm-12 col-xl-8`}>
                <h5><Link to={`/res/${props.resId}`}>{props.resName}</Link></h5>
                <h5>{props.name} - ${props.price}</h5>
                <p>{props.description}</p>
            </div>
            <div className={`MenuItem-Buttons col-12 col-sm-12 col-xl-2`}>
            </div>
        </div>
        </>

    )
}   

export default FoodSearchMenuItem;