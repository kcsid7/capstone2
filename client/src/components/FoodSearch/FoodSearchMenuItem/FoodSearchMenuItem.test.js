import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom'; 

import FoodSearchMenuItem from './FoodSearchMenuItem';


test("Renders the Food Search Menu Item properly", () => {
    render(<BrowserRouter><FoodSearchMenuItem /></BrowserRouter>);
})


test("FoodSearchMenuItem snapshot", () => {
    const { asFragment } = render(<BrowserRouter><FoodSearchMenuItem /></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
})