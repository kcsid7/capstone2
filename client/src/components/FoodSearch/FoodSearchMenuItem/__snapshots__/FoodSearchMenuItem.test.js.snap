// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodSearchMenuItem snapshot 1`] = `
<DocumentFragment>
  <div
    class="MenuItem container row"
  >
    <div
      class="MenuItem-Image col-12 col-sm-12 col-xl-2"
    />
    <div
      class="MenuItem-Details col-12 col-sm-12 col-xl-8"
    >
      <h5>
        <a
          href="/res/undefined"
        />
      </h5>
      <h5>
         - $
      </h5>
      <p />
    </div>
    <div
      class="MenuItem-Buttons col-12 col-sm-12 col-xl-2"
    />
  </div>
</DocumentFragment>
`;
